addons/mrp/models/mrp_production.py

```diff
@@ -697,7 +697,7 @@ def _search_delay_alert_date(self, operator, value):
     @api.depends('company_id', 'date_start', 'is_planned', 'product_id')
     def _compute_date_finished(self):
         for production in self:
-            if not production.date_start or production.is_planned:
+            if not production.date_start or production.is_planned or production.state == 'done':
```

Comment:
I would add state in compute depends and remove the force write in action_done()

Reply:
not sure what you mean 
the write is actually necessary to set date_finished as now (at the moment the button is clicked)

Reply:
yes you are right